src/chaat/errors.clj

```diff
@@ -1,16 +1,40 @@
 (ns chaat.errors)
 
 (defn do-or-error
-  "Checks the result of a previous operation before performing the next 
-   operation. Will be used in a let block. If there is an error, the error will 
-   cascade down the remaining expressions in the let block without performing 
-   the remaining operations.
+  " Used to check the result of a previous operation before performing the next
+operation. To be used in a let block. If there is an error, the error will
+cascade down the remaining expressions in the let block without executing the
+remaining operations.
 
-   In situations where function f does not return a result map, it is usually 
-   when the operation is not a validation or db query. Here, one may not be 
-   concerned about checking that particular output of f before continuing, and 
-   we use the result map of the previous validation/query to continue the chain."
+fn f is required to return a result map with structure {:result res :error err}
+
+In situations where function f does not return a result map, it is usually
+when the operation is not a validation or db query. Here, you may not be
+concerned about checking that particular output of f before continuing, and
+we use the result map of the previous validation/query to continue the chain. "
   [{:keys [result error]} f & args]
-  (if (nil? error)
+  (if-not error
     (apply f args)
     {:result result :error error}))
+
+;; This is a work-in-progress macro to improve the do-or-error flow.
+;; You can place a :result key in the args lists to an expression to pass in the 
+;; :result key value from the result map of the previous expression.
+
+(defn do-result-or-error
+  " Same as do-or-error but allows for the placement of an optional :result key
+to pass the :result key value from the result map of the previous expression "
+  [{:keys [result error]} f & args]
+  (if-not error
+    (apply f (map #(if (= % :result) result %) args))
```

Comment:
question: What is the anonymous fn `#(if (= % :result) result %)` intended to do? 

Also consider extracting this to a private fn with a name that makes this more explicit

Reply:
`(map #(if (= % :result) result %) args)` iterates over items in `args` and replaces `:result` with `result`. In this scope, the `:result` key value from the result map was deconstructed into `result`.
This allows us to place a `:result` key in the args list to an expression which essentially passes in the `:result` key value from the result map of the previous threaded expression. 

This is written in a confusing way, and the naming makes it more confusing. It was mainly an experiment to see if we could improve my clunky `do-or-error` flow. If chaat was to continue, this flow would likely be replaced by a library or simplified.

I'll be happy to discuss why this felt necessary at the time the next time we speak.